express/src/middleware/mobileValidator.spec.ts

```diff
@@ -0,0 +1,142 @@
+   })
+ 
+   it('A valid +44 number with spaces will be cleaned into one without. ', async function () {
+       assert.equal(removeSpaceAndDash(VALID_NUMBER_STARTING_WITH_PLUS_44_CONTAINING_SPACES), VALID_NUMBER_STARTING_WITH_PLUS_44, `Expected ${VALID_NUMBER_STARTING_WITH_PLUS_44_CONTAINING_SPACES} to be ${VALID_NUMBER_STARTING_WITH_PLUS_44}`);
+   })
+ 
+ 
+   it('A valid 07 number with dashes will be cleaned into one without. ', async function () {
+       assert.equal(removeSpaceAndDash(VALID_NUMBER_STARTING_WITH_0_CONTAINING_DASHES), VALID_NUMBER_STARTING_WITH_0, `Expected ${VALID_NUMBER_STARTING_WITH_0_CONTAINING_DASHES} to be ${VALID_NUMBER_STARTING_WITH_0}`);
+   })
+ 
+   it('A valid +44 number with dashes will be cleaned into one without. ', async function () {
+       assert.equal(removeSpaceAndDash(VALID_NUMBER_STARTING_WITH_PLUS_44_CONTAINING_DASHES), VALID_NUMBER_STARTING_WITH_PLUS_44, `Expected ${VALID_NUMBER_STARTING_WITH_PLUS_44_CONTAINING_DASHES} to be ${VALID_NUMBER_STARTING_WITH_PLUS_44}`);
+   })
+ 
+   it('A valid 07 number without spaces will not be cleaned ', async function () {
+       assert.equal(removeSpaceAndDash(VALID_NUMBER_STARTING_WITH_0), VALID_NUMBER_STARTING_WITH_0, `Expected ${VALID_NUMBER_STARTING_WITH_0} to be ${VALID_NUMBER_STARTING_WITH_0}`);
+   })
+ 
+   it('A valid +44 number without spaces will not be cleaned . ', async function () {
+       assert.equal(removeSpaceAndDash(VALID_NUMBER_STARTING_WITH_PLUS_44), VALID_NUMBER_STARTING_WITH_PLUS_44, `Expected ${VALID_NUMBER_STARTING_WITH_PLUS_44} to be ${VALID_NUMBER_STARTING_WITH_PLUS_44}`);
+   })
+ 
+})
+ 
+describe('Checking that only numbers starting with +447[1-9] and 07[1-9] are accepted', () => {
+   it('An invalid +48 number is rejected.', async function () {
+       assert.equal(await ukNumberCheck(INVALID_NUMBER_STARTING_WITH_PLUS_48), false, `${INVALID_NUMBER_STARTING_WITH_PLUS_48} should not be allowed as it does not begin with +447`);
+   })
+ 
+   it('A valid +44 number is accepted.', async function () {
+       assert.equal(await ukNumberCheck(NUMBER_BEGINNING_WITH_PLUS_4470), true, `${NUMBER_BEGINNING_WITH_PLUS_4470} should be allowed as it does not begin with +447`);
+   })
+ 
+   it('An invalid 08 number is rejected.', async function () {
+       assert.equal(await ukNumberCheck(NUMBER_BEGINNING_WITH_08), false, `${NUMBER_BEGINNING_WITH_08} should not be allowed as it does not begin with +447`);
+   })
+ 
+   it('An valid 07 number is rejected.', async function () {
+       assert.equal(await ukNumberCheck(INVALID_NUMBER_STARTING_WITH_PLUS_44), true, `${INVALID_NUMBER_STARTING_WITH_PLUS_44} should not be allowed as it does not begin with +447`);
+   })
+  
+})
+ 
+ 
+ 
+describe('Checking that only numbers starting with +447[1-9] and 07[1-9] are accepted', () => {
```

Comment:
This describe line is copy pasted from above - if the tests inside it fit the description, we don't need another describe line.  If they don't then the line should be unique and describe what the tests are doing.